scripts/ASTImportTest.sh

```diff
@@ -1,105 +1,174 @@
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) solidity contributors.
+# ------------------------------------------------------------------------------
+# Bash script to test the import/exports.
+# ast import/export tests:
+#   - first exporting a .sol file to JSON, then loading it into the compiler
+#     and exporting it again. The second JSON should be identical to the first.
+
+set -euo pipefail
 
-set -e
-
-# Bash script to test the ast-import option of the compiler by
-# first exporting a .sol file to JSON, then loading it into the compiler
-# and exporting it again. The second JSON should be identical to the first
 READLINK=readlink
 if [[ "$OSTYPE" == "darwin"* ]]; then
     READLINK=greadlink
 fi
 REPO_ROOT=$(${READLINK} -f "$(dirname "$0")"/..)
 SOLIDITY_BUILD_DIR=${SOLIDITY_BUILD_DIR:-${REPO_ROOT}/build}
-SOLC=${SOLIDITY_BUILD_DIR}/solc/solc
-SPLITSOURCES=${REPO_ROOT}/scripts/splitSources.py
+SOLC=$SOLIDITY_BUILD_DIR/solc/solc
+SPLITSOURCES=$REPO_ROOT/scripts/splitSources.py
+
+# shellcheck source=scripts/common.sh
+source "${REPO_ROOT}/scripts/common.sh"
+
+function print_usage
+{
+  fail "Usage: ${0} ast [--exit-on-error]."
+}
+
+function print_used_commands
+{
+    local test_directory="$1"
+    local export_command="$2"
+    local import_command="$3"
+    printError "You can find the files used for this test here: ${test_directory}"
+    printError "Used commands for test:"
+    printError "# export"
+    echo "$ ${export_command}" >&2
+    printError "# import"
+    echo "$ ${import_command}" >&2
+}
+
+function print_stderr_stdout
+{
+    local error_message="$1"
+    local stderr_file="$2"
+    local stdout_file="$3"
+    printError "${error_message}"
+    printError ""
+    printError "stderr:"
+    cat "${stderr_file}" >&2
+    printError ""
+    printError "stdout:"
+    cat "${stdout_file}" >&2
+}
+
+IMPORT_TEST_TYPE=
+EXIT_ON_ERROR=0
+for PARAM in "$@"
+do
+  case "$PARAM" in
+    ast) IMPORT_TEST_TYPE="ast" ;;
+    --exit-on-error) EXIT_ON_ERROR=1 ;;
+    *) print_usage ;;
+  esac
+done
 
 SYNTAXTESTS_DIR="${REPO_ROOT}/test/libsolidity/syntaxTests"
 ASTJSONTESTS_DIR="${REPO_ROOT}/test/libsolidity/ASTJSON"
-NSOURCES="$(find "$SYNTAXTESTS_DIR" -type f | wc -l)"
-
-# DEV_DIR="${REPO_ROOT}/../tmp/contracts/"
-# NSOURCES="$(find $DEV_DIR -type f | wc -l)" #TODO use find command
 
 FAILED=0
 UNCOMPILABLE=0
 TESTED=0
 
-if [[ "$(find . -maxdepth 0 -type d -empty)" == "" ]]; then
-    echo "Test directory not empty. Skipping!"
-    exit 1
-fi
+function ast_import_export_equivalence
+{
+    local sol_file="$1"
+    local input_files=( "${@:2}" )
+
+    # export ast - save ast json as expected result (silently)
+    local export_command="$SOLC --combined-json ast --pretty-json --json-indent 4 ${input_files[*]} > expected.json 2> stderr_export.txt"
+    if ! $SOLC --combined-json ast --pretty-json --json-indent 4 "${input_files[@]}" > expected.json 2> stderr_export.txt
```

Comment:
I was not able to find a way to just reuse `$export_command` here somehow. Any ideas? @r0qs @cameel

Reply:
Doing this should work (but with some caveats):
'```bash
$export_command > expected.json 2> stderr_export.txt
```
- First, the redirects must be outside of the variable.
- Second, shellcheck will complain that it's unquoted.

Something like this is probably better:
```bash
local export_command=("$SOLC" --combined-json ast --pretty-json --json-indent 4 "${input_files[@]}")
"${export_command[@]}"
```

Now, though the problem is that you cannot easily pass arrays into functions (at least not without them just being converted to a simple string).

I think that the best solution might be to make it a function and instead pass the name of that function around.